megamek/src/megamek/common/verifier/TestMech.java

```diff
@@ -1449,6 +1452,22 @@ public boolean hasIllegalEquipmentCombinations(StringBuffer buff) {
             }
         }
         
+        if (mech.hasWorkingWeapon(WeaponType.F_HYPER)) {
+            switch (mech.hasEngine() ? mech.getEngine().getEngineType() : Engine.NONE) {
```

Comment:
Since FUSION is the only legitimate engine type, I would recommend just checking for that instead, rather than all of the other engine types (case Engine.FUSION: break, default: illegal)

Reply:
Checking for a fusion engine is common enough that there's a convenience method for it.
`if (mech.hasWorkingWeapon(WeaponType.F_HYPER) && !(mech.hasEngine() && mech.getEngine().isFusion()) {
  //yada yada
}